docs/user_guide/Upgrade_SIMP/Version_Maps/6.3.3_6.4.0.inc

```diff
@@ -71,3 +79,152 @@ functionality:
+
+Update the repos for the OS and SIMP and run yum update
+
+The following errors/warnings are expected:
+
+.. code-block:: bash
+
+  warning: /var/simp/environments/simp/FakeCA/togen saved as /var/simp/environments/simp/FakeCA/togen.rpmsave
+  warning: /var/simp/environments/simp/FakeCA/ca.cnf saved as /var/simp/environments/simp/FakeCA/ca.cnf.rpmsave
+  libsemanage.semanage_direct_remove_key: Unable to remove module simp-environment at priority 400. (No such file or directory).
+  /usr/sbin/semodule:  Failed!
+
+If you are using the ``FakeCA`` in the ``simp`` environment, follow the steps in the ``FakeCA`` section above to restore the needed files.
+
+The old selinux module, ``simp-environment``, had to be removed before installing the
+the new selinux module, ``simp-environment-rsync`` to avoid conflicts when the
+simp-environment-skeleton module is installed.  Therefore
+the error above is given when the simp-environment RPM tries to delete it.
+
+Create the Puppetfile
+""""""""""""""""""""""
+At this point in the upgrade the puppet modules have been installed as git repositories
+under ``/usr/share/simp/modules``.  Now the Puppetfile, used by r10k to deploy modules
+to the environment, needs to be created.
+
+The environment name in this example is ``myenv``.  Run the following commands to create a default Puppetfile.
+
+.. code-block:: bash
+
+  # Change to the top level puppet environment directory for your environment.
+  cd /etc/puppetlabs/code/environments/myenv
+  # Generate the main Puppetfile
+  simp puppetfile generate --skeleton > Puppetfile
+  # Generate the Puppetfile for the simp modules.
+  simp puppetfile generate > Puppetfile.simp
+
+Edit the main Puppetfile and make sure all the local modules you created
+and have in the modules directory are listed here.   This includes the ``site`` module,
+which is no longer delivered.
+
+.. WARNING::
+
+  Any module not listed in Puppetfile and the files it includes will be deleted from the
+  modules directory.
+
+The following shows 3 examples of how to include your modules in the Puppetfile:
+
+*  site: a module that is not a git repo and resides in the modules directory
+*  mymodule: a git repo that resides in a directory on the puppet server
+*  mysrvmod: a git repo on a gitlab/github type server.
+
+.. code-block:: yaml
+
+  mod 'site',
+    :local => true
+
+  mod 'mymodule'
+    :git => '/usr/share/mymodules/mymodule',
+    :tag => '1.1.1'
+
+  mod 'mysrvmod'
+    :git => 'https://gitserver.my.domain/mygitproject/mysrvmod.git'
+    :tag => '1.0.1'
+
+.. IMPORTANT::
+
+   Make sure all the  modules under your environments modules directory
+   that you created and are not simp modules are listed  in the Puppetfile.
+   This includes the ``site`` module, which is no longer delivered.
+
+   Any module not listed in the Puppetfile will be deleted from the modules 
+   directory along with the files it includes.
+   modules directory.
+
+Deploy the Puppet Modules to the Environment
```

Comment:
- Should we use the term "install" instead of "deploy", because this is not using ``r10k deploy``?
- Do we want to refer to some section about control repos and installing via ``r10k install``?
- This section only applies to folks who do not use r10k already...should that be made more clear?

Reply:
I like the term deploy instead of install because "install" is more of an RPM term.
We are deploying the modules, just using the puppetfile instead of a control repo to deploy.